profile.md

```diff
@@ -170,6 +170,22 @@ One item not addressed in detail in this document is how the issuer decides on w
    <td>User/VO
    </td>
   </tr>
+  <tr>
+   <td><strong>Virtual Organization</strong>
+   </td>
+   <td>A Virtual Organization (VO) is a grouping of people and/or IT services that is defined only in the specific IT service used to manage the VO and that serves to allow grid resources to be shared just between its members.<br/><br/>
+
+By construction, the members of a VO work towards common goals through the use of grid resources to which they have access. For example:
+- the processing and analysis of data produced by an experiment;
+- the provision of shared services to supported projects.
+
+A VO typically has an internal structure that may be exposed to some extent, such as (sub)groups to which VO members may belong and roles they may take on for specific activities on grid resources.
+
+Each VO has its own membership criteria and each grid site makes its own decisions on which resources to make available to which VOs.
+   </td>
+   <td>LHC experiment VOs
+   </td>
+  </tr>
```

Comment:
This definition and text is very much focussed on the original ideas of VOs in Grid and WLCG. I feel this would make it very difficult to adopt the profile outside of only the 4 WLCG VOs. Do we really need such an extensive definition of a VO?

Reply:
Would it be any better if the definition removed all mention of the word "grid" (e.g., "grid resources" becomes "resources") ?

Do you have a counter-example: something you would consider a VO but would not comfortably fit the definition?  That would help focus on the parts are causing the trouble and would need to be rephrased.

Reply:
I think it makes good sense to avoid grid references where they are unnecessary or even unwanted and have adjusted the text accordingly in this commit:

https://github.com/WLCG-AuthZ-WG/common-jwt-profile/pull/47/commits/57be70aacce54dc75deeca6743ddd2f4a1698e02

What do you think?

Reply:
I agree it's better, but I think it would be good to have input from e.g. @NicolasLiampotis, @jbasney and @DrDaveD whether this also is a good match for many of the small VOs managed by EGI and likewise for what are called COs by COmanage and typically called Communities by AARC.

I'd say it would at least be an entity responsible for
1. providing an attribute authority (both for group/role membership of its members and for providing the authorizations they have (capabilities)).
2. handle accounting
3. have MoUs etc. with sites on use of resources (not sure we can say they own any resources: they own data and have rights to CPU time etc.)
There are probably a few more points needed.

Reply:
I think this describes the fermilab VO fine.  Our subVOs could be considered subgroups as described.

Reply:
Hi Mischa,
I do not think we need to say anything about items 2 and 3 in this document, as those would, to a large extent, be implementation details that may differ significantly between VOs and/or their resource providers.  For example, a Tier-3 site in the wider WLCG ecosystem (i.e. beyond the official MoU sites) does not have to provide its own accounting.  A VO will typically have its own accounting for its own reasons, which would then cover resource usage not otherwise accounted for, but I do not think there exists a requirement for that either.  Furthermore, accounting comes after the fact: it is not needed to get things done in the first place.  A Tier-3 site in WLCG may have an MoU directly with its VO(s), but there also exists resource usage, not always at small scales, that is not covered by any MoU.  For example, a resource provider may simply configure support for a VO without wanting to go through an expensive process to formalize the arrangement, when an informal agreement is deemed sufficient.

Reply:
Concerning accounting, this is of course also depending on the issue of whether there's VO<>issuer correspondence, but in any case, when a job comes in, a site would like to be able to tell on behalf of which VO it is running? So the token should be able to provide some way of telling that? And as such it should be part of the profile I think, in the sense that we should provide the means to a site or service to have the necessary information?

Reply:
ATM, the token issuer is deemed to denote the VO, for which an externally defined mapping then needs to be configured on each service that needs to handle tokens from the given VO:

issuer URL --> VO

We would like to have more flexibility there and hopefully we will get that as an outcome of the GUT Profile WG...

Reply:
@maarten-litmaath I agree with your sentiment (issuer URL --> VO); however please note that issue #28 is currently still open.

Moreover, I think @msalle has a point: that accounting could be mentioned as a concrete use-case, one that drives the need to identify through which VO a service provider's resources are being used/consumed.

Such accounting makes sense because we anticipate MoUs (or SLAs, perhaps) being agreed between the service provider and the VO.  This, in turn, provides some expectation on what is a VO (that they are able to negotiate access to resources).

These expectations might be best articulated by including them in the document's description of a VO.

Reply:
It is not a sentiment, the description of the `iss` claim in the Common Claims subsection says:

`For WLCG this would be the VO.`

I now think we probably should indeed mention accounting as a concrete use case for the notion of a VO. To be continued...